File: 1768_merge_strings_alternately.py

def mergeAlternately(word1: str, word2: str) -> str:
    smaller_len = len(word1) if len(word1) < len(word2) else len(word2)
    new_str = ""

    for i in range(smaller_len):
        new_str += word1[i]
        new_str += word2[i]

    if i < len(word1) - 1:
        new_str += word1[i+1:]
    elif i < len(word2) - 1:
        new_str += word2[i+1:]

    return new_str

File: go/1768_merge_strings_alternately.go
package main

func mergeAlternately(word1 string, word2 string) string {
	merged := ""

	shorterStr := word1
	longerStr := word2
	if len(word2) < len(word1) {
		shorterStr = word2
		longerStr = word1
	}

	i := 0
	for i < len(shorterStr) {
		merged = merged + string(word1[i])
		merged = merged + string(word2[i])
		i++
	}

	if len(word1) != len(word2) {
		merged = merged + string(longerStr[i:])
	}

	return merged
}
